pkg/zimage/zimage.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package zimage contains a Parser for the arm zImage Linux format. It assumes
+// little endian arm.
+package zimage
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+)
+
+// Magic values used in the zImage header and table.
+const (
+	Magic         = 0x016f2818
+	Endianess     = 0x04030201
+	TableMagic    = 0x45454545
+	TagKernelSize = 0x5a534c4b
+)
+
+// ZImage is one of the major formats used by Linux on ARM. This struct
+// is only for storing the metadata.
+type ZImage struct {
+	Header Header
+	Table  []TableEntry
+}
+
+// Header appears near the beginning of the zImage.
+//
+// The layout is defined in Linux:
+//     arch/arm/boot/compressed/head.S
+type Header struct {
+	Magic      uint32 `offset:"0x0"`
+	Start      uint32 `offset:"0x4"`
+	End        uint32 `offset:"0x8"`
+	Endianess  uint32 `offset:"0xc"`
+	TableMagic uint32 `offset:"0x10"`
+	TableAddr  uint32 `offset:"0x14"`
+}
+
+// Table is an extension to Header. A zImage may have 0 or more entries.
+//
+// The layout is defined in Linux:
+//     arch/arm/boot/compressed/vmlinux.lds.S
+type TableEntry struct {
+	Tag  uint32
+	Data []uint32
+}
+
+// Parse a ZImage from a file.
+func Parse(f *os.File) (*ZImage, error) {
```

Comment:
I think io.ReadSeeker is better then os.File
Then you will be able to create simple tests.

Reply:
Done!